packages/marketplace-management/src/components/ui/apps/app-referrals-display.tsx

```diff
@@ -0,0 +1,66 @@
+import { InstallationModel, ReferralTypeModelPagedResult } from '@reapit/foundations-ts-definitions'
+import React, { FC, memo, useMemo } from 'react'
+import { BodyText } from '@reapit/elements'
+
+import { useReapitGet } from '@reapit/utils-react'
+import { reapitConnectBrowserSession } from '../../../core/connect-session'
+import { GetActionNames, getActions } from '@reapit/utils-common'
+
+interface ReferralsDisplayProps {
+  installations: InstallationModel[]
+  hasCurrentInstallations: boolean
+}
+
+/**
+ * This is not ideal at all but not as expensive as it looks. In practice, there is only ever one active installation
+ * so I only have to access the inner reduce funcs for a single installation as a return earlt in the first loop.
+ * I then need to loop through meta data and then through allow array to match the ids to the referral type names.
+ * Horrible but little other choice.
+ */
+
+export const handleReferralText =
```

Comment:
finding it difficult to follow this function to make sure it's fool proof. Don't suppose you'd mind add some extra test cases against it to double check it's savvy?